OnLand/ControlLaptop/angular-UI/src/app/components/display-log/display-log.component.ts

```diff
@@ -0,0 +1,147 @@
+import { Component, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { logData } from 'src/app/services/models';
+import { DisplayLogService } from 'src/app/services/display-log/display-log.service';
+import { stringify } from '@angular/compiler/src/util';
+@Component({
+  selector: 'app-display-log',
+  templateUrl: './display-log.component.html',
+  styleUrls: ['./display-log.component.css']
+})
+
+export class DisplayLogComponent implements OnInit {
+
+  private telemetrySubscriber: Subscription;
+
+  logArchive: logData;
+  logDummy=[];
+
+  // note: assuming machine initialises with neutral state
+  constructor(private displayLogService:DisplayLogService) { 
+    this.logArchive = {
+      advancing: false,
+      retracting: false,
+      neutral: true,
+      cutterheadValve: 0,
+      augerValve: 0
+    }
+  }
+
+  ngOnInit(): void {
+    this.telemetrySubscriber = this.displayLogService.getLogs().subscribe
+    ((msg: { message: string }) => {
+      console.log('beep boop server says ' + JSON.stringify(msg));
+
+      switch (parseInt(msg.message)) {
+
+        // cases 0-2 is the status of the TBM (boolean only)
+        case 0:
+          this.advance();
+          break;
+          
+        case 1:
+          this.retract();
+          break;
+        case 2:
+          this.neutral();
+          break;
+
+        // cases 3-4 are the valves (0-1), math round and random is passing a fake 2-decimal value 
+        case 3:
+          this.cutterheadValve( Math.round( Math.random() *100 )/100 ) 
+          this.logDummy.push("$: Cutterhead Needle Valve -> " + this.logArchive.cutterheadValve );
+          break;
+        case 4:
+          this.augerValve( Math.round( Math.random() *100 )/100 );
+          this.logDummy.push("$: Auger Needle Valve-> " + this.logArchive.augerValve );
+          break;
+
+        default:
+          //just dummy text
+          this.logDummy.push( "$: nothing" );
+      }
+    });
+  }
+
+  ngOnDestroy() {
+    this.telemetrySubscriber.unsubscribe();
+  }
+
+
+  // function calls to update status of TBM
+
+  // BOOLEAN ONLY STATUSES
+  // assuming only one of the three (adv/ret/neu) can be true at a given instance 
+
+  // A/R/N are almost duplicates, they print any changes or lack thereof
+  advance() {
+    if (this.logArchive.advancing == true){
```

Comment:
in javascript, it is always recommended to use [strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) instead of [equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality). 
More reading can refer to: https://basarat.gitbook.io/typescript/recap/equality

Reply:
so please use ===